src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -296,10 +296,25 @@ public async Task<Stream> GetStreamingFileAsync(long version, Partition partitio
         });
     }
 
+    /// <inheritdoc />
+    public async Task<long> GetFilePropertiesContentLengthAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
```

Comment:
Should this instead be an extension method that unifies both `IFileStore` and `IInstanceStore`? Essentially we want to join `GetInstanceIdentifierWithPropertiesAsync` and `GetFilePropertiesAsync`

Reply:
Things I can think of off the top of my head:
- Some new abstraction that leverages both `IFileStore` and `IInstanceStore`
- An extension method on `IInstanceStore` that does `GetInstanceIdentifierWithPropertiesAsync` with an `IFileStore` as an argument
- An extension method of `FileProperties` that uses an `IFileStore`

Reply:
Sounds good to me!